harvard-university-of-worcester.csl

```diff
@@ -0,0 +1,376 @@
+          <text variable="locator"/>
+        </group>
+      </group>
+    </layout>
+  </citation>
+  <bibliography et-al-use-first="4" et-al-min="5" hanging-indent="true">
+    <sort>
+      <key macro="author"/>
+      <key variable="title"/>
+    </sort>
+    <layout>
+      <choose>
+        <if match="any" variable="author">
+          <text macro="author"/>
+        </if>
+        <else-if type="broadcast" match="all" variable="container-title">
+          <text macro="single-quote-title"/>
+        </else-if>
+        <else-if match="any" variable="editor">
+          <text macro="editor"/>
+        </else-if>
+        <else>
+          <text macro="italicised-title" suffix="."/>
+        </else>
+      </choose>
+      <text macro="year-date-bibliography" prefix=" (" suffix=")"/>
+      <choose>
+        <if match="all" variable="author collection-title">
+          <text macro="single-quote-title"/>
+        </if>
+        <else-if match="any" variable="author editor">
+          <choose>
+            <if type="chapter article-magazine article-newspaper post-weblog" match="any">
+              <text macro="standard-title" prefix=" " suffix="."/>
+            </if>
+            <else>
+              <text macro="italicised-title" prefix=" " suffix="."/>
+            </else>
+          </choose>
+        </else-if>
+      </choose>
+      <text macro="original-title"/>
+      <text macro="scale" prefix=" " suffix="."/>
+      <text macro="director"/>
+      <text macro="interviewer"/>
+      <text macro="translator" prefix=" "/>
+      <choose>
+        <if type="chapter" match="any">
+          <text term="in" text-case="capitalize-first" suffix=":"/>
+        </if>
+        <else-if type="paper-conference" match="all" variable="container-title">
+          <text term="in" text-case="capitalize-first" suffix=": "/>
+        </else-if>
+      </choose>
+      <choose>
+        <if match="any" variable="author">
+          <text macro="editor" prefix=" "/>
+        </if>
+      </choose>
+      <text macro="container-title" prefix=" " suffix="."/>
+      <text macro="collection-title"/>
+      <text macro="series-episode"/>
+      <text macro="edition" prefix=" " suffix="."/>
+      <text macro="genre"/>
+      <text macro="receipient" prefix=" " suffix=","/>
+      <text macro="broadcast-details" prefix=" "/>
+      <text macro="day-month"/>
+      <text macro="medium-online" prefix=" "/>
+      <group>
+        <text macro="volume-number" prefix=" " suffix=","/>
+        <text macro="issue-number" prefix=" " suffix="."/>
+      </group>
+      <group>
+        <text macro="event-name"/>
+        <choose>
+          <if match="none" variable="publisher-place">
+            <text macro="event-place"/>
+          </if>
+        </choose>
+        <text macro="event-date"/>
+      </group>
+      <text macro="publisher-place" prefix=" "/>
+      <text macro="pages" prefix=" " suffix="."/>
+      <choose>
```

Comment:
again, try to find a way -- likely a group -- to solve this without putting punctuation into value fields.